python/lsst/cp/pipe/utils.py

```diff
@@ -391,7 +407,7 @@ def errFunc(p, x, y):
         randomDelta = np.random.normal(0., sigmaErrTotal, len(dataY))
         randomDataY = dataY + randomDelta
         randomFit, _ = leastsq(errFunc, initialParams,
-                               args=(dataX, randomDataY), full_output=0)
+                               args=(dataX, randomDataY, 1./np.sqrt(randomDataY)), full_output=0)
```

Comment:
Seems like `randomDataY` can be negative here - what happens then? I think you end up with a `RuntimeWarning` and a bunch of `nan`s in the return array, and then I'm not sure what happens downstream...